mesonbuild/compilers.py

```diff
@@ -1053,7 +1053,7 @@ def compiles(self, code, env, extra_args=None, dependencies=None, mode='compile'
     def _links_wrapper(self, code, env, extra_args, dependencies):
         "Shares common code between self.links and self.run"
         args = self._get_compiler_check_args(env, extra_args, dependencies, mode='link')
-        return self.compile(code, args.to_native())
+        return self.compile(code, args)
```

Comment:
Yes, I thought maybe it was called on a subset of the args, but it looks like it's called on all of it.